test/spec/modules/liveIntentIdHubSystem_spec.js

```diff
@@ -0,0 +1,419 @@
+      consentString: 'consentDataString'
+    })
+    gppConsentDataStub.returns({
+      gppString: 'gppConsentDataString',
+      applicableSections: [1, 2]
+    })
+    liveIntentIdHubSubmodule.getId(defaultConfigParams).callback(() => {});
+
+    const expectedConsent = { gdpr: { consentString: 'consentDataString', gdprApplies: true }, gpp: { applicableSections: [1, 2], consentString: 'gppConsentDataString' }, usPrivacy: { consentString: '1YNY' } }
+
+    expect(window.liQHub).to.have.length(3)
+
+    expect(window.liQHub[0]).to.eql({
+      clientDetails: { name: 'prebid', version: '$prebid.version$' },
+      clientRef: {},
+      collectSettings: { timeout: DEFAULT_AJAX_TIMEOUT },
+      consent: expectedConsent,
+      integration: { distributorId: defaultConfigParams.distributorId, publisherId: '89899', type: 'custom' },
+      partnerCookies: new Set(),
+      resolveSettings: { identityPartner: 'prebid', timeout: DEFAULT_AJAX_TIMEOUT },
+      type: 'register_client'
+    })
+
+    expect(window.liQHub[1]).to.eql({
+      clientRef: {},
+      delay: defaultConfigParams.params.fireEventDelay,
+      type: 'schedule_default_collect'
+    })
+
+    const resolveCommand = window.liQHub[2]
+    // functions cannot be reasonably compared, remove them
+    delete resolveCommand.onSuccess[0].callback
+    delete resolveCommand.onFailure
+
+    expect(resolveCommand).to.eql({
+      clientRef: {},
+      onSuccess: [{ type: 'callback' }],
+      requestedAttributes: [ 'nonId' ],
+      type: 'resolve'
+    })
+  });
+
+  it('should fire an event when getId and a hash is provided', function() {
+    liveIntentIdHubSubmodule.getId({ params: {
+      ...defaultConfigParams.params,
+      emailHash: '58131bc547fb87af94cebdaf3102321f'
+    }}).callback(() => {});
+
+    expect(window.liQHub).to.have.length(3)
+
+    expect(window.liQHub[0]).to.eql({
+      clientDetails: { name: 'prebid', version: '$prebid.version$' },
+      clientRef: {},
+      collectSettings: { timeout: DEFAULT_AJAX_TIMEOUT },
+      consent: {},
+      integration: { distributorId: defaultConfigParams.distributorId, publisherId: '89899', type: 'custom' },
+      partnerCookies: new Set(),
+      resolveSettings: { identityPartner: 'prebid', timeout: DEFAULT_AJAX_TIMEOUT },
+      type: 'register_client'
+    })
+
+    expect(window.liQHub[1]).to.eql({
+      clientRef: {},
+      sourceEvent: { hash: '58131bc547fb87af94cebdaf3102321f' },
+      type: 'collect'
+    })
+
+    const resolveCommand = window.liQHub[2]
+    // functions cannot be reasonably compared, remove them
+    delete resolveCommand.onSuccess[0].callback
+    delete resolveCommand.onFailure
+
+    expect(resolveCommand).to.eql({
+      clientRef: {},
+      onSuccess: [{ type: 'callback' }],
+      requestedAttributes: [ 'nonId' ],
+      type: 'resolve'
+    })
+  });
+
+  it('should have the same data after call decode when appId, disrtributorId and sourceEvent is absent', function() {
```

Comment:
The test says "when appId, disrtributorId and sourceEvent is absent", can it be made more explicit in line 124? E.g. by setting the `distributorId` explicitly to undefined instead of relying on `defaultConfigParams.distributorId` being undefined?

Reply:
Changed [here](https://github.com/LiveIntent/Prebid.js/pull/42/files#diff-fd86a0ed5dfa75fa7316fc905719b9bb5ff9460873b893dc0cd07d04e58586c4R150)